### PR TITLE
feat: add makefile dictionary

### PR DESCRIPTION
**Pull request by kevintraver**
https://github.com/streetsidesoftware/cspell-dicts/issues/2665

### CODE REVIEW

**Comment by calvinballing** [comment]
Beat me to it! 😄  Let me see if the list I started has anything not on yours

**Comment by calvinballing** [comment]
I haven't gone through this term by term, but I went to this page: https://www.gnu.org/software/make/manual/make.html
And pulled out the terms that cspell identified as typos.

Here are the extra terms I found:
addsuffix,alloca,altmake,archivefile,ARFLAGS,ASFLAGS,Automake,backquotes,bastante,bigoutput,bindir,binprefix,bish,BISONFLAGS,blish,chgrp,cleanall,cleandiff,cleanobj,CMDFLAGS,COFLAGS,COMSPEC,conditionalizing,CTANGLE,CWEAVE,CXXFLAGS,datadir,datarootdir,DBSD,DDEF,DDEFBLOCKING,DDIRENT,DDOPRNT,DEFBLOCKING,DEFFLAGS,DEMUL,DESTDIR,DFTIME,DFTRUNCATE,diffarch,dira,dirb,DJGPP,DMKDIR,docdir,doprnt,DPOSIX,DRENAME,DSIGTYPE,DSTDC,DSTRSTR,DUSG,DVALLOC,dvidir,DVPRINTF,DXENIX,errno,escrito,Esta,etags,extradeps,fcntl,Feldman,FFLAGS,filea,fileb,filenm,fixincludes,fname,foolib,fprintf,frob,frobnicate,frobnicating,frobozz,frobs,ftime,ftruncate,funcs,getcwd,getdate,getoldopt,getopt,GFLAGS,globbing,Gnulib,gnumake,GNUMAKEFLAGS,Grath,GUILEIO,headerfile,htmldir,Ibar,Ifoo,includedir,incrementation,infodir,installcheck,INSTALLDATA,installdirs,intcmp,ioctls,Isrc,Jobserver,lcurses,LDCONFIG,LDLIBS,LFLAGS,lfoo,lgnu,lgnumake,libcurses,libdir,libexec,libexecdir,libfoo,libgcc,libgnumake,libname,LIBPATTERNS,LINTFLAGS,lispdir,littleoutput,lname,LOADLIBES,localedir,localstatedir,magtape,MAKEINFO,MAKELEVEL,MAKEOVERRIDES,MAKESHELL,mandir,manext,mfile,MFLAGS,mkclose,mkfifo,mkinstalldirs,mknod,mkopen,MKPORT,mkstemp,mkwrite,MODFLAGS,mostlyclean,MTIO,Multiauthor,myfile,myprog,myrule,ndir,NOEXPAND,nombre,nondirectory,NOTINTERMEDIATE,NOTPARALLEL,nullstring,OBJDIR,oldincludedir,Oline,onefile,ONESHELL,ONEVAR,oneword,pathnames,pathsearch,pdfdir,PFLAGS,POSIX,PREPROCESS,preprocessable,prereq,printenv,psdir,RANLIB,Ratfor,realclean,RECIPEPREFIX,recompiles,remsh,rescanned,REXEC,RFLAGS,RTAPELIB,runstatedir,sbindir,SECONDEXPANSION,sedscript,shar,sharedstatedir,SHELLFLAGS,SHELLSTATUS,sinclude,srcdir,SRCS,Stallman,stdlib,strcat,strcpy,strerror,strlen,strstr,struct,subtargets,SYMDEF,sysconfdir,sysmacros,tcexparg,TERMERR,TERMOUT,testpad,TEXI,Texinfo,TMPDIR,twofile,TWOVAR,txinfo,uncatchable,Undefining,unexport,uninstallation,unistd,unstripped,valloc,vprintf,wildmat,YFLAGS,yourpkg

**Comment by calvinballing** [comment]
I don't want to step on your toes if you are planning any other changes.  Do you want to add the extra terms or should I?  I haven't pulled your branch down yet, but I can

**Comment by kevintraver** [comment]
> I don't want to step on your toes if you are planning any other changes. Do you want to add the extra terms or should I? I haven't pulled your branch down yet, but I can

Just added them! Thanks!

**Comment by Jason3S** [comment]
You guys rock!

File: dictionaries/makefile/cspell-ext.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    // Dictionaries to always be used.
+    // Generally left empty
+    "dictionaries": [],
+    // Language Rules to apply to matching files.
+    // Files are matched on `languageId` and `local`
+    "languageSettings": [
+        {
+            // VSCode languageId. i.e. typescript, java, go, cpp, javascript, markdown, latex
+            // * will match against any file type.
+            "languageId": "*",
```
Comment (Jason3S) [inline dictionaries/makefile/cspell-ext.json:25]:
```suggestion
            "languageId": "makefile",
```

**Review by Jason3S — COMMENTED** [review]
Thank you.

File: dictionaries/makefile/package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "access": "public"
+  },
+  "exports": {
+    ".": "./cspell-ext.json",
+    "./cspell": "./cspell-ext.json",
+    "./cspell-ext.json": "./cspell-ext.json"
+  },
+  "scripts": {
+    "build": "cross-env NODE_OPTIONS=--max_old_space_size=8192 cspell-tools-cli build",
+    "test": "head -n 1000 \"src/makefile.txt\" | cspell -v -c ./cspell-ext.json \"--local=*\" \"--languageId=*\" stdin",
```
Comment (Jason3S) [inline dictionaries/makefile/package.json:16]:
```suggestion
    "test": "head -n 1000 \"src/makefile.txt\" | cspell -v -c ./cspell-ext.json \"--local=*\" \"--languageId=makefile\" stdin",
```

**Pull request merged**